src/filters.rs

```diff
@@ -67,6 +69,17 @@ pub struct ExternalFilter {
     pub argument: Option<Argument>,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct FirstFilter {
+    pub at: SourceSpan,
```

Comment:
It doesn't really matter, but I'd keep `at` as a `(usize, usize)` until using it in `RenderError`:

```suggestion
    pub at: (usize, usize),
```

The reason is basically just to avoid using the third party type (`miette::SourceSpan`) until it's necessary.